packages/riipen-ui/src/components/Badge.test.jsx

```diff
@@ -0,0 +1,310 @@
+      const anchor = { horizontal: "left", vertical: "bottom" };
+
+      const wrapper = mount(<Badge anchor={anchor} />);
+
+      expect(
+        wrapper
+          .find("Badge")
+          .childAt(0)
+          .childAt(0)
+          .hasClass("bottom-left-rectangle")
+      ).toEqual(true);
+    });
+
+    it("gives an error when given an invalid anchor", () => {
+      const errors = jest.spyOn(console, "error").mockImplementation();
+      const anchor = { horizontal: "side", vertical: "up" };
+
+      mount(<Badge anchor={anchor} />);
+
+      expect(errors).toHaveBeenCalledTimes(1);
+    });
+  });
+
+  describe("children prop", () => {
+    it("displays given children", () => {
+      const child = <div>child</div>;
+
+      const wrapper = mount(<Badge>{child}</Badge>);
+
+      expect(wrapper.containsMatchingElement(child)).toBeTruthy();
+    });
+  });
+
+  describe("classes prop", () => {
+    it("sets custom class", () => {
+      const classVariant = ["newClass"];
+
+      const wrapper = mount(<Badge classes={classVariant} />);
+
+      expect(
+        wrapper
+          .find("Badge")
+          .childAt(0)
+          .childAt(0)
+          .hasClass(classVariant[0])
+      ).toEqual(true);
+    });
+  });
+
+  describe("color prop", () => {
+    it("sets color class name with a valid color", () => {
+      const color = "muted";
+
+      const wrapper = mount(<Badge color={color} />);
+
+      expect(
+        wrapper
+          .find("Badge")
+          .childAt(0)
+          .childAt(0)
+          .hasClass(color)
+      ).toEqual(true);
+    });
+
+    it("gives an error when given an invalid color", () => {
+      const errors = jest.spyOn(console, "error").mockImplementation();
+
+      mount(<Badge color="blue" />);
+
+      expect(errors).toHaveBeenCalledTimes(1);
+    });
+  });
+
+  describe("component prop", () => {
+    it("sets component with valid custom component", () => {
+      const component = "a";
+
+      const wrapper = mount(<Badge component={component} />);
+
+      expect(wrapper.find(component)).toHaveLength(1);
+      expect(
+        wrapper
+          .find("Badge")
+          .childAt(0)
+          .name()
+      ).toEqual(component);
+    });
+
+    it("gives an error when given an invalid component", () => {
+      const errors = jest.spyOn(console, "error").mockImplementation();
+      const component = "abcdefg";
+
+      mount(<Badge component={component} />);
+
+      expect(errors).toHaveBeenCalledTimes(1);
+    });
+  });
+
+  describe("content prop", () => {
+    it("badge renders when content is 0 and showZero is false", () => {
+      const content = 0;
+
+      const wrapper = mount(<Badge content={content} />);
+
+      expect(
+        wrapper
+          .find("Badge")
+          .childAt(0)
+          .childAt(0)
+          .hasClass("hidden")
+      ).toEqual(true);
+      expect(Number(wrapper.text())).toEqual(content);
+    });
```

Comment:
Slightly confused. According to the component, the "0" should not render if showZero is `false` (which is the default`. so why is line 163 actually working? Maybe it's because it actually _does_ render, but the CSS is hiding it.

I would maybe recommend just deleting line 163 as it's sort of confusing and rename the test maybe to "it hides the content when content is 0 and showZero is false".

Does that make sense?

Reply:
Yes it does